app/src/androidTest/java/ch/epfl/sdp/appart/CreateUserAccountUITest.java

```diff
@@ -0,0 +1,102 @@
+package ch.epfl.sdp.appart;
+
+
+import androidx.test.espresso.action.ViewActions;
+import androidx.test.espresso.intent.Intents;
+import androidx.test.espresso.matcher.ViewMatchers;
+import androidx.test.ext.junit.rules.ActivityScenarioRule;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.util.concurrent.ExecutionException;
+
+import javax.inject.Inject;
+
+import ch.epfl.sdp.appart.login.LoginService;
+import ch.epfl.sdp.appart.user.CreateUserActivity;
+import ch.epfl.sdp.appart.user.LoginActivity;
+import ch.epfl.sdp.appart.user.User;
+import dagger.hilt.android.testing.HiltAndroidRule;
+import dagger.hilt.android.testing.HiltAndroidTest;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.action.ViewActions.typeText;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.intent.Intents.intended;
+import static androidx.test.espresso.intent.matcher.IntentMatchers.hasComponent;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+@HiltAndroidTest
+public class CreateUserAccountUITest {
+
+    @Rule(order = 0)
+    public HiltAndroidRule hiltRule = new HiltAndroidRule(this);
+
+    @Rule(order = 1)
+    public ActivityScenarioRule<CreateUserActivity> createUserActivityActivityRule = new ActivityScenarioRule<>(CreateUserActivity.class);
+
+    @Inject
+    LoginService loginService;
+
+    @Before
+    public void init() {
+        hiltRule.inject();
+        Intents.init();
+        loginService.useEmulator("10.0.2.2", 9099);

```

Comment:
We should really mock the login service in the futur

Reply:
Yes maybe, but I feel like the emulator makes it useless. It's like duplicating code. But I understand that it would be better. I will add it in another PR.